docs/posts/roadmap-2024-H1/index.qmd

```diff
@@ -0,0 +1,219 @@
+---
+title: "Ibis project 2024 roadmap"
+author: "Cody Peterson"
+date: "2024-02-10"
+image: commits.png
+categories:
+    - blog
+    - roadmap
+    - community
+---
+
+## Overview
+
+Welcome to the first public roadmap for the Ibis project! If you aren't familiar
+with the background of Ibis or who supports it nowadays, we recommend reading
+[why Voltron Data supports Ibis](../why-voda-supports-ibis/index.qmd) before the
+roadmap below.
+
+## 2024 roadmap
+
+We have a [public roadmap as a GitHub
+project!](https://github.com/orgs/ibis-project/projects/5)
+
+![Ibis roadmap](roadmap.png)
+
+We are early in our use of this GitHub project, so please pardon any
+disorganization as we get it up and running efficiently. In general, we have:
+
+- **Roadmap view**: consisting of meta-issues in their respective repositories
+ for high-level objectives of the Ibis project
+- **Triage view**: consisting of new issues across Ibis project repositories
+ that need to be triaged
+- **Backlog view**: consisting of issues that have been triaged (assigned a
+ priority) and are on the backlog
+- **TODO view**: consisting of issues that are in progress or ready to be worked
+ on soon
+- **Label-specific views**: consisting of issues for specific labels, like
+ documentation or a large refactor
+
+Right now, [the team at Voltron Data](../why-voda-supports-ibis/index.qmd) sets
+the roadmap and priorities. Over time as more contributors and organizations
+join the project, we expect this process to diversify and become more
+community-driven.
+
+### Overall themes
+
+Our top five themes for 2024 include:
+
+1. **Ibis backends**: Ibis is a Python frontend for many backends. To continue
+scaling to more backends, we need to complete a major rework of library
+internals and stabilize the API for backend authors. Related work in this area
+will make it easier than ever to create new Ibis backends and maintain them.
+This work will also include improving backend interfaces for operations like
+table creation, insertion, and upsertion.
+
+2. **Ibis for ML**: Increasingly, data projects are ML projects. Ibis can
+uniquely help with feature engineering and other ML tasks connecting your data
+where it lives to ML models. We will continue to improve Ibis for ML use cases.
+
+3. **Ibis for streaming data**: Ibis has only been for batch data until very
+recently. With the addition of the first streaming backends, we will continue to
+improve Ibis for streaming data use cases and bridge the gap between batch and
+streaming data.
+
+4. **Ibis for geospatial**: Ibis has a rich set of geospatial expressions, but
+most backends do not implement them. We will continue to improve Ibis for
+geospatial use cases and bridge the gap between geospatial data and other data
+types.
+
+5. **Ibis community**: Ibis is an open source project and we want to make it as
```

Comment:
Should we include a point to differentiate us from other open-core community? We need to build strong trust in the community over time, and it starts with messaging.

Reply:
I don't think we need to split hairs on the minutia of open source versus open core versus open standards. Both open core and open standards are open source, but they only make sense in the context of a product and Ibis isn't a product. I think the distinction is useful, but it belongs at the level of a given product not here.